### PR TITLE
[JUJU-2919] Merge decoupled api onto develop

### PR DESCRIPTION
**Pull request by cderici**
Merges the `decouple-apiclient` branch into `develop`.

### QA Steps

Please confirm the source and target, i.e. double check that the right stuff is going into the right place.

### CODE REVIEW

File: rpc/params/firewall.go [hunk 1]
```diff
@@ -58,5 +58,5 @@ func (v KnownServiceValue) Validate() error {
 	case SSHRule, JujuControllerRule, JujuApplicationOfferRule:
 		return nil
 	}
-	return errors.NotValidf("known service %q", v)
```
Comment (hmlanigan) [inline rpc/params/firewall.go:61]: where did this change come from?

**Review by hmlanigan — COMMENTED** [review]
Just the one question, and a unit test is failing.

File: rpc/params/firewall.go (hunk 1 above)
Comment (cderici) [reply rpc/params/firewall.go:61]: I managed to pull it out of my own records, apparently it's coming from this one (from September 2022) https://github.com/juju/juju/pull/14608, looks like we didn't like the resulting error message https://github.com/juju/juju/pull/14608#discussion_r970814743

**Comment by cderici** [comment]
> Just the one question, and a unit test is failing.

Why's that test (`TestSetEntitiesAnnotation`) is passing for me locally?

**Comment by hmlanigan** [comment]
> Why's that test (`TestSetEntitiesAnnotation`) is passing for me locally?

Looks like the input to annotations.Set is a map, and may change order. Try running the test with the race script. It may take some interactions to fail.

https://github.com/juju/juju/wiki/Stress-Test

**Pull request merged**